datavault/backend.py

```diff
@@ -51,24 +91,205 @@ def size(self):
+            self.comments = [getComment(i) for i in range(count)]
+        else:
+            self.comments = []
+
+    def save(self):
+        S = util.DVSafeConfigParser()
+
+        sec = 'General'
+        S.add_section(sec)
+        S.set(sec, 'Created',  time_to_str(self.created))
+        S.set(sec, 'Accessed', time_to_str(self.accessed))
+        S.set(sec, 'Modified', time_to_str(self.modified))
+        S.set(sec, 'Title',       self.title)
+        S.set(sec, 'Independent', repr(len(self.independents)))
+        S.set(sec, 'Dependent',   repr(len(self.dependents)))
+        S.set(sec, 'Parameters',  repr(len(self.parameters)))
+        S.set(sec, 'Comments',    repr(len(self.comments)))
+
+        for i, ind in enumerate(self.independents):
+            sec = 'Independent {}'.format(i+1)
+            S.add_section(sec)
+            S.set(sec, 'Label', ind.label)
+            S.set(sec, 'Units', ind.unit)
+
+        for i, dep in enumerate(self.dependents):
+            sec = 'Dependent {}'.format(i+1)
+            S.add_section(sec)
+            S.set(sec, 'Label',    dep.legend)
+            S.set(sec, 'Units',    dep.unit)
+            S.set(sec, 'Category', dep.label)
+
+        for i, par in enumerate(self.parameters):
+            sec = 'Parameter {}'.format(i+1)
+            S.add_section(sec)
+            S.set(sec, 'Label', par['label'])
+            # encode the parameter value as a data-url
+            data_url = labrad_urlencode(par['data'])
+            S.set(sec, 'Data', data_url)
+
+        sec = 'Comments'
+        S.add_section(sec)
+        for i, (time, user, comment) in enumerate(self.comments):
+            time = time_to_str(time)
+            S.set(sec, 'c{}'.format(i), repr((time, user, comment)))
+
+        with open(self.infofile, 'w') as f:
+            S.write(f)
+
+    def initialize_info(self, title, indep, dep):
+        self.title = title
+        self.accessed = self.modified = self.created = datetime.datetime.now()
+        self.independents = indep
+        self.dependents = dep
+        self.parameters = []
+        self.comments = []
+        self.cols = len(indep) + len(dep)
+
+    @property
+    def dtype(self):
+        return np.dtype(','.join(['f8']*self.cols))
+
+    def access(self):
+        self.accessed = datetime.datetime.now()
+
+    def getIndependents(self):
+        return self.independents
+
+    def getDependents(self):
+        return self.dependents
+
+    def getRowType(self):
+        units = []
+        for var in self.independents + self.dependents:
+            units.append('v[{}]'.format(var.unit))
+        type_tag = '*({})'.format(','.join(units))
+        return type_tag
+
+    def getTransposeType(self):
+        units = []
+        for var in self.independents + self.dependents:
+            units.append('*v[{}]'.format(var.unit))
+        type_tag = '({})'.format(','.join(units))
+        return type_tag
+
+    def addParam(self, name, data):
```

Comment:
This does not do any checking of data for labrad types whereas HDF5MetaData does. This could cause a breakage if some weird data type was being stored using addParam.